cloudbuild.yaml

```diff
@@ -9,6 +9,7 @@ steps:
     - PROW_GIT_TAG=$_GIT_TAG
     - PULL_BASE_REF=$_PULL_BASE_REF
     - VERSION=$_PULL_BASE_REF
+    - HOME=/root  # for docker buildx
```

Comment:
Looks like this is needed only if we use docker buildx. Is that the case?

Reply:
`buildx build` was used for dnsmasq image. And I expected that changing homedir should work for both cases (using buildx and build)

But I've updated `rules.mk` to use buildx as well for consistency.